apps/charterafrica/src/lib/ecosystem/airtable/processData.js

```diff
@@ -0,0 +1,194 @@
+  const {
+    schema: { partnerTableColumns },
+  } = config;
+  const { name, url, logo } = partnerTableColumns;
+  const { partners } = tableData;
+  const mapped = supporters.map((id) => {
+    try {
+      const { fields } = partners.find((item) => id === item.id);
+      return {
+        name: getter(fields, name),
+        website: getter(fields, url),
+        logo: getter(fields, logo),
+      };
+    } catch (error) {
+      Sentry.captureMessage(error.message);
+      return null;
+    }
+  });
+  return mapped.filter(Boolean);
+}
+
+function mapSocialMediaToFields(socialMedia, config, tableData) {
+  const {
+    schema: { socialMediaTableColumns },
+  } = config;
+  const { name, url } = socialMediaTableColumns;
+  const { socialMedia: socialMediaData } = tableData;
+  const mapped = socialMedia.map((id) => {
+    try {
+      const { fields } = socialMediaData.find((item) => item.id === id);
+      return {
+        name: getter(fields, name),
+        website: getter(fields, url),
+      };
+    } catch (error) {
+      Sentry.captureMessage(error.message);
+      return null;
+    }
+  });
+  return mapped.filter(Boolean);
+}
+
+export function processTool(data, config, tableData) {
+  try {
+    const {
+      schema: { toolTableColumns },
+    } = config;
+    const theme = {
+      en: getter(data, toolTableColumns.theme.en)?.[0],
+      pt: getter(data, toolTableColumns.theme.pt)?.[0],
+      fr: getter(data, toolTableColumns.theme.fr)?.[0],
+    };
+    const description = {
+      en: getter(data, toolTableColumns.description.en),
+      pt: getter(data, toolTableColumns.description.pt),
+      fr: getter(data, toolTableColumns.description.fr),
+    };
+
+    const operatingCountries = getter(
+      data,
+      toolTableColumns.operatingCountries
+    );
+    const homeCountry = getter(data, toolTableColumns.homeCountry);
+    const partners = mapSupportersToFields(
+      getter(data, toolTableColumns.partners) || [],
+      config,
+      tableData
+    );
+    const supporters = mapSupportersToFields(
+      getter(data, toolTableColumns.supporters) || [],
+      config,
+      tableData
+    );
+    const socialMedia = mapSocialMediaToFields(
+      getter(data, toolTableColumns.socialMedia) || [],
+      config,
+      tableData
+    );
+    return {
+      airtableId: data.id,
+      avatarUrl: getter(data, toolTableColumns.avatarUrl)?.[0]?.url,
+      externalId: getter(data, toolTableColumns.slug) || " ",
+      repoLink: getter(data, toolTableColumns.source.url),
+      name: getter(data, toolTableColumns.name),
+      link: getter(data, toolTableColumns.url),
+      operatingCountries,
+      contributors: getter(data, toolTableColumns.contributors),
+      supporters,
+      partners,
+      homeCountry,
+      socialMedia,
+      theme,
+      description,
+    };
+  } catch (error) {
+    Sentry.captureMessage(error.message);
+    return null;
+  }
```

Comment:
We're not making any external calls, not sure where we expect these errors to come from.

Reply:
I think it is important to catch errors at every level to avoid termination if subsequent processes @kilemensi

Reply:
No, it's not important to catch errors @koechkevin, it's important to handle errors. It's more important to understand why errors are thrown and if possible, write code that doesn't run into exerptions.

Hence, if we've pulled all data from Airtable already, what kind of errors can be thrown here?